Style the "Awaiting Grade" notice

donnapep opened the pull request:
Resolves https://github.com/Automattic/sensei/issues/7074.

## Proposed Changes
- Updates the styles for the "Awaiting Grade" notice on the quiz page.
- Although the scope was only for the notice on the quiz page, I also updated just the colors of the "Awaiting Grade" notice on the lesson page. Since I broke it with my changes, the easiest fix was to match the colors. Matching the fonts would also be nice for consistency's sake, but that is outside the scope of this PR.
- Unfortunately, the HTML of the _Pending Teacher Grade_ and the _Restart Quiz_ link in the notice and in the quiz footer are different. We probably should have synced up better on this one. Perhaps one day soon we'll be able to match the markup and use common styles. 🙏🏻 
- Made a change to ensure the third-party theme's `learning-mode.css` file loads after the generic Sensei one. Now we won't have to make theme CSS selectors more specific than the default selectors in order to override them. Instead, we just need to match them, and the styles loaded last (i.e. the theme's styles) win.

## Testing Instructions
<!--
Add as many details as possible to help others reproduce the issue and test the changes.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Activate Divi or Astra.
2. Add a quiz to a lesson and ensure that quiz is *not* set to auto-grade.
3. Take the course as a student.
4. View the "Awaiting grade" notice on the quiz page and ensure it matches the design.
5. Verify that the colors of the "Awaiting grade" notice on the lesson page matches that of the quiz page (it's expected that the fonts won't match).
6. Test on mobile as well.
7. Check out this branch - https://github.com/Automattic/themes/pull/7401.
8. Activate the Course theme.
9. Repeat steps 4 & 5 for all style variations.
10. Disable the Quiz Graded email and view the notice again.
11. Verify that "You’ll receive an email once it’s ready to view." is no longer part of the message.
12. Bonus: Disable the `email_customization` feature flag and repeat step 11.

## Pre-Merge Checklist
<!-- Complete applicable items on this checklist **before** merging. Items that are not applicable can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed. -->
- [x] PR title and description contain sufficient detail and accurately describe the changes
- [x] Acceptance criteria is met
- [x] Decisions are publicly documented
- [x] Adheres to coding standards ([PHP](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/), [JavaScript](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/), [CSS](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/), [HTML](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/html/))
- [x] All strings are translatable (without concatenation, handles plurals)
- [x] Follows our naming conventions (P6rkRX-4oA-p2)
- [ ] Hooks (p6rkRX-1uS-p2) and functions are documented
- [x] New UIs are responsive and use a [mobile-first approach](https://zellwk.com/blog/how-to-write-mobile-first-css/)
- [x] New UIs match the designs
- [ ] Different user privileges (admin, teacher, subscriber) are tested as appropriate
- [x] Code is tested on the minimum supported PHP and WordPress versions
- [x] User interface changes have been tested on the latest versions of Chrome, Firefox and Safari
- [ ] "Needs Documentation" label is added if this change requires updates to documentation
- [ ] Known issues are created as new GitHub issues

donnapep committed:
Ensure theme-specific LM CSS styles load after Sensei's

This change is specifically to load the third-party theme's `learning-mode.css` file after the generic Sensei one.

Now we won't have to make theme CSS selectors more specific than the default selectors in order to override them. Instead, we just need to match them, and the styles loaded last (i.e. the theme's styles) will win.

donnapep committed:
Tweak the markup for the notice

donnapep committed:
Style "Awaiting Grade" notice

donnapep committed:
Add change log

donnapep committed:
Use the same colors for the notice on the lesson page

donnapep committed:
Merge branch 'trunk' into update/awaiting-grade-notice

donnapep committed:
Minor tweaks to spacing and font

donnapep committed:
Fix linter

donnapep committed:
Show default cursor when hovering on "Pending Teacher Grade"

donnapep committed:
Ensure message is conditional based on whether graded email is enabled

donnapep committed:
Ensure array is declared

donnapep committed:
Merge branch 'trunk' into update/awaiting-grade-notice

Imran92 reviewed (commented):
Looks nice to me 👍 I've found a few small issues, not all of them are for addressing, as some looks like design inconsistencies to me-

1. The text inside the notice is in two lines in the desktop version. I don't think it's important, but just mentioning as I've noticed this. Also looks like it's inconsistent among the variations
Current:
<img width="972" alt="image" src="https://github.com/Automattic/sensei/assets/6820724/99914f70-d9be-44ca-9fc5-399820c49607">

Design:
<img width="1262" alt="image" src="https://github.com/Automattic/sensei/assets/6820724/23efbc46-2959-48d2-b9ad-23285c0b973f">

2. The notice in the mobile view is full-width in the design, but currently it has lateral spacings
Current:
<img width="503" alt="image" src="https://github.com/Automattic/sensei/assets/6820724/bd5714c8-a82b-4608-9a16-5a1d3a7eebd9">

Design:
<img width="503" alt="image" src="https://github.com/Automattic/sensei/assets/6820724/fed4ee8f-9726-4bc3-9048-6971afcc2542">

3. The notice has a 4px border-radius in the design
Current:
<img width="296" alt="image" src="https://github.com/Automattic/sensei/assets/6820724/4f4b6aad-d72b-492c-ae84-04476b06f414">

Design:
<img width="647" alt="image" src="https://github.com/Automattic/sensei/assets/6820724/19dc20f3-cd0f-4f3b-8bb6-d0b682f4692e">

4. In the mobile view, the design has only 24px spacing below the restart button, in our case, the 24px gets added up with the button spacing and adds more space below the button text
Current:
<img width="466" alt="Screenshot 2023-10-04 at 6 40 46 PM" src="https://github.com/Automattic/sensei/assets/6820724/a0053c6c-90a8-498a-919f-4b31228d9e7e">

Design:
<img width="709" alt="image" src="https://github.com/Automattic/sensei/assets/6820724/b2ce88ad-10c9-43a6-af41-d0f1b7cd1265">

donnapep committed:
Add border radius to Course theme notice

donnapep committed:
Add border radius only on desktop

donnapep committed:
Make notice full width on mobile

donnapep committed:
Use less bottom padding in mobile view

donnapep commented:
> The text inside the notice is in two lines in the desktop version. I don't think it's important, but just mentioning as I've noticed this. Also looks like it's inconsistent among the variations

Yup, this one felt like a design inconsistency. I opted not to use a line break because none of the mobile views have it, and it didn't feel like a good use of my time considering the very low impact.

> The notice in the mobile view is full-width in the design, but currently it has lateral spacings

Fixed in https://github.com/Automattic/sensei/pull/7197/commits/56a373a9b9caba919c88cf4c134a0e094a5734ae. I think we should keep the button width consistent with what you've done in the footer, which is not full-width. WDYT?

> The notice has a 4px border-radius in the design

Fixed here for just the Course theme - https://github.com/Automattic/sensei/pull/7197/commits/dbbdbdc23424c1ec6694542612259b2ce9930f6e

> In the mobile view, the design has only 24px spacing below the restart button, in our case, the 24px gets added up with the button spacing and adds more space below the button text

Fixed here - https://github.com/Automattic/sensei/pull/7197/commits/6aa4ba711b5e0e7ada2ff5aeafccfec42cedb426

@Imran92

Imran92 reviewed (dismissed):
> I think we should keep the button width consistent with what you've done in the footer, which is not full-width. WDYT?

Yap! it looks good to me too.

As we are rendering a button here and changing text based on different conditions, should we also add some tests to make sure they're rendering as expected and only when they should? But maybe we can also add the tests with the "Graded" notice when we'll have the full UI ready for all the states and conditions. I'm good with both.

I've also noticed that the "Pending teacher grade" button is being rendered in the graded quiz notice too while it should render only for quizzes awaiting grade. I'm good with that being handled in the "Graded" notice PR, but only a bit concerned as we're merging this PR to trunk and trunk can go to release, in which case it may result in a UI bug. It won't be an issue if we also merge the Graded notice PR before the next release.

<img width="1072" alt="Screenshot 2023-10-05 at 6 25 54 PM" src="https://github.com/Automattic/sensei/assets/6820724/7a2e9a5a-f96d-4bb9-b88b-e932195c0655">

I'm approving the PR as it looks great otherwise and in case we decide to take care of the concerns in another PR 👍 🚀

donnapep commented:
> I've also noticed that the "Pending teacher grade" button is being rendered in the graded quiz notice too while it should render only for quizzes awaiting grade. 

Yes, I noticed this as well. My plan is not to merge this PR until the graded quiz notice is ready too. I think I'll add tests with that PR as well.

donnapep committed:
Merge branch 'trunk' into update/awaiting-grade-notice

donnapep committed:
Refactor button function to return markup similar to WP button

Also accept CSS classes as a parameter

donnapep committed:
Refactor "Pending teacher grade" button to use shared function

donnapep committed:
Add "Continue to next lesson" button for graded quizzes

donnapep committed:
Eliminate quiz-graded.scss and move code to notices.scss

All the in quiz-graded.scss if for notices anyway

donnapep committed:
Move notice-specific CSS from buttons.scss to notices.scss

donnapep committed:
Style "Continue to Next Lesson" button

donnapep committed:
Fix some styling issues

It got to be too time-consuming trying to add either 8px or 24px of bottom padding on mobile depending on whether the Restart Quiz link was visible or not, so I simplified and made the bottom padding be 24px in both cases.

donnapep committed:
Don't show next lesson button if the user failed the quiz

donnapep committed:
Right-align "Restart Quiz" link

donnapep committed:
Display "Contact Teacher" button under certain conditions

donnapep committed:
Fix PHP linter

donnapep committed:
Add change log entry

donnapep committed:
Add tests for get_primary_button_html function

donnapep committed:
Change quiz grade notice title to an `h2`

donnapep committed:
Move some CSS to enable Course's `theme.json` to be applied more easily

donnapep committed:
Change tests back from confirming they would fail :laughing:

donnapep commented:
@Imran92 

> As we are rendering a button here and changing text based on different conditions, should we also add some tests to make sure they're rendering as expected and only when they should?

I don't think we should be writing unit tests to test UI. We should use e2e tests for that. However, we currently don't have any e2e tests for quizzes 😅 , so I think that bit may need to be part of a larger initiative. I did write some unit tests in https://github.com/Automattic/sensei/pull/7210 for a function that you originally wrote and I refactored.

donnapep committed:
Merge branch 'trunk' into update/awaiting-grade-notice

donnapep committed:
Eliminate quiz-graded.scss and move code to notices.scss

donnapep committed:
Add some whitespace between title and text

donnapep committed:
Merge branch 'update/awaiting-grade-notice' into update/graded-notice

donnapep committed:
Fix $lesson_status not defined

donnapep committed:
Remove href if empty so button is not clickable

Also update the cursor style on hover.

donnapep committed:
Allow null for $url parameter

renatho commented on assets/css/sensei-course-theme/notices.scss:
```diff
@@ -1,22 +1,24 @@
-	border-color: var(--border-color);
-	color: inherit;
-	padding: 1.125rem 1.5rem;
+	background-color: var(--wp--preset--color--primary, #f6f7f7);
```
One thing I noticed while reviewing the other PR based on this one is that maybe this background could be a little weird depending on the theme colors.

Here is what happens with the TT3:

<img width="924" alt="Screenshot 2023-10-20 at 15 36 20" src="https://github.com/Automattic/sensei/assets/876340/818a280d-2fe7-4f54-9ece-6e35b69dc7e1">

I tried to think of a suggestion here, but I couldn't think of any suggestion that wouldn't be to have a default color not dependent on the theme. And maybe override it specifically in the Course theme with the color we want?

donnapep replied:
@renatho Ohhhh...interesting! In https://github.com/Automattic/sensei/pull/7197/commits/196fd9644a3afb1648e984bf3621ded20566bd24, I ended up switching to use the foreground color. For the Course theme, primary is the same color as foreground. Now, if a block theme sets both the foreground and background colors, they should work well together. There may still be a small number of cases where it doesn't work for some themes, but as long as it's not a top Sensei theme, I think it's fine.

renatho replied:
Thank you @donnapep! And the color looks good to me on TT3 now. :)

donnapep committed:
Switch to using foreground color

donnapep committed:
Merge branch 'trunk' into update/awaiting-grade-notice

donnapep committed:
Update assets/css/sensei-course-theme/notices.scss

Co-authored-by: Renatho De Carli Rosa <renatho@gmail.com>

donnapep committed:
Fix style issue on Twenty Twenty Three

donnapep committed:
Merge pull request #7210 from Automattic/update/graded-notice

Style the "Your Grade" notice

donnapep committed:
Merge branch 'trunk' into update/awaiting-grade-notice

Imran92 reviewed (approved):
LGTM 👍 

I've faced some issues when we have the latest Gutenberg enabled. The theme styles weren't getting applied as expected.

#### Without Gutenberg on
![Screenshot 2023-11-01 at 2 52 42 PM](https://github.com/Automattic/sensei/assets/6820724/95417235-19b6-4f48-8d9b-e1befb988d4b)

#### With Gutenberg on
![Screenshot 2023-11-01 at 2 52 13 PM](https://github.com/Automattic/sensei/assets/6820724/1a538523-f5c4-4d10-9d50-fab00eb46ed5)

I'll go ahead and approve it. We can address the issue with Gutenberg either here or separately

donnapep commented:
> I'll go ahead and approve it. We can address the issue with Gutenberg either here or separately

Separately I think, given the notices look pretty bad in Production currently. 🙂
